retrofit/src/main/java/retrofit/Utils.java

```diff
@@ -39,6 +39,12 @@
     return object;
   }
 
+  static boolean methodRequiresRequestBody(final String method) {
```

Comment:
No. Also this doesn't handle arbitrary HTTP methods defined by the user with `hasBody = true`.